Px4 firmware nuttx 7.29+ sdio fixes

davids5 opened the pull request:
This PR addressees hangs when the SD card is removed while writing.

https://github.com/PX4/Firmware/issues/12071#issuecomment-542681971
https://github.com/PX4/Firmware/issues/13087#issuecomment-538345550

nuttx commented:
Committed: https://bitbucket.org/patacongo/nuttx/commits/f326ebd276fcac19a831867f41af6683a234b594

This was a difficult merge.  You will probably want to verify the result.

Greg

davids5 committed:
[BACKPORT] stm32: SDIO Fix system hang on card eject.

davids5 committed:
[BACKPORT] stm32f7:SDMMC Fix system hang on card eject.

gregory-nutt committed:
[BACKPORT] stm32h7: SDMMC Fix system hang on card eject.

davids5 committed:
[BACKPORT] stm32l4_sdmmc.c:  SDMMC Fix system hang on card eject.

davids5 committed:
[BACKPORT] mmcsd_sdio.c:  Set wrbusy after success of all the operations.

   Any failure could leave wrbusy set when it should not be if the operation failed.

davids5 committed:
[BACKPORT] mmcsd_sdio.c: Enter a removed card state On a failure.

   If we fail to read cardstatus in mmcsd_transferready consider this a hard error and set the card to removed.

davids5 commented:
@PX4/testflights - Please test on FMUv4 and FMUv5
